### PR TITLE
[RSPEC MODEL] User model rspec

### PR DESCRIPTION
**Pull request by huytn-1183**
## Related Tickets
- [#43132](https://edu-redmine.sun-asterisk.vn/issues/43132)

## WHAT (optional)
- validate
- enum define
- digest generate
- token generate
- attribute truthy
- authenticated?

## HOW
- RSpect model user
- Gem shoulda

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by huytn-1183** [comment]
ready

**Comment by vanvtt-0952** [comment]
commented
lại..conflict

**Comment by huytn-1183** [comment]
ready

**Comment by vanvtt-0952** [comment]
commented
Conflict code nè cu (leuleu)

**Comment by huytn-1183** [comment]
ready

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  context "validate" do
+    it { should validate_presence_of(:name) }
+    it { should validate_length_of(:password) }
+    it { should validate_presence_of(:email) }
+    it { should validate_length_of(:email) }
+    it { should validate_uniqueness_of(:email) }
+  end
+
+  it ".digest not duplicate" do
```
Comment (vanvtt-0952) [inline spec/models/user_spec.rb:18]: Mi đang test cho gem bcypt, cái này thì ko cần, nó đưa ra gem là nó test rồi :v

**Comment by vanvtt-0952** [comment]
commented

**Comment by huytn-1183** [comment]
ready

**Comment by huytn-1183** [comment]
ready

**Pull request merged**